### ISSUE TITLE
🤖 [AUTO] Implement Strategy: roe_bm_factor_market_neutral_quintile_prediction

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#532]
## 🤖 Auto-Generated Strategy Request

**Source:** Machine Trading by Ernest P. Chan.pdf (strategy 5)
**Extracted:** 2025-12-27T05:35:49.084312

---

1.  **Strategy Name**: roe_bm_factor_market_neutral_quintile_prediction
2.  **Strategy Type**: Factor-based, Market-neutral, Monthly prediction
3.  **Timeframe**: Monthly (hold for one month, monthly rebalancing)
4.  **Instruments**: SPX stocks (large-cap equities)

5.  **Entry Rules**:
    *   Compute ROE and BM factor loadings as defined in the `roe_bm_factor_net_long_prediction` strategy.
    *   Exclude stocks with non-positive ROE or BM.
    *   Fit a linear regression model on a training set using `log(ROE)` and `log(BM)` to predict next-month returns.
    *   **Long Entry**: Rank predicted returns for all stocks. Long stocks in the top quintile of predicted returns.
    *   **Short Entry**: Short stocks in the bottom quintile of predicted returns.
    *   **Condition**: Only enter positions if there are five or more stocks with earnings announcements in the previous day, ensuring sufficient valid data for quintile sorting.

6.  **Exit Rules**:
    *   **Time-based**: Hold positions for one month. Rebalance all positions monthly based on new predictions.
    *   *Take Profit Conditions*: Not specified.
    *   *Stop Loss Conditions*: Not specified.

7.  **Indicators Required**:
    *   ROE (Return on Equity)
    *   BM (Book-to-Market Ratio)

8.  **Risk Management**:
    *   **Position Sizing**: Equal weighting for stocks within the top/bottom quintiles to maintain market neutrality.
    *   *Maximum Risk per Trade*: Not specified.
    *   *Risk-Reward Ratio*: Not specified.

9.  **Additional Notes**:
    *   This market-neutral variant of the ROE and BM factor model generated negative returns (CAGR of -9.2%) on the test set.

---

## Implementation Instructions

Please implement this strategy using the `backtesting.py` framework with the following requirements:

1. **File Location:** `strategies/roe_bm_factor_market_neutral_quintile_prediction.py`
2. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting
3. **Indicators:** Use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators)
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
